src/Tutor.Web/Controllers/Domain/KCController.cs

```diff
@@ -15,11 +16,13 @@ public class KCController : ControllerBase
     {
         private readonly IMapper _mapper;
         private readonly IKCService _kcService;
+        private readonly IInstructor _instructor;
 
-        public KCController(IMapper mapper, IKCService kcService)
+        public KCController(IMapper mapper, IKCService kcService, IInstructor instructor)
         {
             _mapper = mapper;
             _kcService = kcService;
+            _instructor = instructor;
```

Comment:
The instructor is a domain service, while controllers usually communicate with application services. You can leave it for now, but we should refactor this. We should consider introducing an instructor service or utilizing existing suitable services. The method of this service should load the data required by the instructor, who can then perform the needed assessments without interacting with a repository. Once done, the instructor returns the result and the app. service reacts by saving what needs to be saved and return what needs to be returned.

For now we can leave it as is.